[Doc] add a doc for sharing a Grafana instance across multiple KubeRay custom resources (#49742)
ray-project/ray: doc/source/cluster/kubernetes/k8s-ecosystem/prometheus-grafana.md

```

* The **install.sh** script creates the above YAML example, [podMonitor.yaml](https://github.com/ray-project/kuberay/blob/master/config/prometheus/podMonitor.yaml#L26-L63) so you don't need to create anything.

> * See the [PodMonitor official document](https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api-reference/api.md#monitoring.coreos.com/v1.PodMonitor) for more details about the configurations.

Reviewer: ```suggestion
* See the official [PodMonitor doc](https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api-reference/api.md#monitoring.coreos.com/v1.PodMonitor) for more details about configurations.
```

* The recording rule above is one of rules defined in [prometheusRules.yaml](https://github.com/ray-project/kuberay/blob/master/config/prometheus/rules/prometheusRules.yaml), and it is created by **install.sh**. Hence, no need to create anything here.

> * See [PrometheusRule official document](https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api-reference/api.md#monitoring.coreos.com/v1.PrometheusRule) for more details about the configurations.

Reviewer: ```suggestion
* See the official [PrometheusRule document](https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api-reference/api.md#monitoring.coreos.com/v1.PrometheusRule) for more details about configurations.
```

* TODO: Note that importing the dashboard manually is not ideal. We should find a way to import the dashboard automatically.

Once the Ray Dashboard is imported into Grafana, you can filter metrics by using the `Cluster` variable. This variable is automatically applied by default when using the provided `PodMonitor` configuration. No additional setup is needed for this labeling.

> If you have multiple RayClusters, the `Cluster` variable allows you to filter metrics specific to a particular cluster. This ensures that you can easily monitor or debug individual RayCluster instances without being overwhelmed by data from all clusters.

Reviewer 2: ```suggestion
If you have multiple RayCluster custom resources, the `Cluster` variable allows you to filter metrics specific to a particular cluster. This ensures that you can easily monitor or debug individual RayCluster instances without being overwhelmed by data from all clusters.
```

> 

Reviewer 2: Add a sentence:

For example, in the following figures, one selects the metrics from the RayCluster `raycluster-embed-grafana`, and the other selects metrics from the RayCluster `raycluster-embed-grafana-2`.

Author: Sure.

![Grafana Ray Dashboard](../images/grafana_ray_dashboard.png)

> Select a specific RayCluster from the dropdown to view metrics for that instance.

Reviewer 2: ```suggestion
```

* TODO: Note that importing the dashboard manually is not ideal. We should find a way to import the dashboard automatically.

## Step 11: View metrics from different RayCluster CRs

> Once the Ray Dashboard is imported into Grafana, you can filter metrics by using the `Cluster` variable. This variable is automatically applied by default when using the provided `PodMonitor` configuration. No additional setup is needed for this labeling.

Reviewer: ```suggestion
Once the Ray Dashboard is imported into Grafana, you can filter metrics by using the `Cluster` variable. Ray Dashboard automatically applies this variable by default when you use the provided `PodMonitor` configuration. You don't need any additional setup for this labeling.
```

> If you have multiple RayCluster custom resources, the `Cluster` variable allows you to filter metrics specific to a particular cluster. This ensures that you can easily monitor or debug individual RayCluster instances without being overwhelmed by data from all clusters.

Reviewer: ```suggestion
If you have multiple RayCluster custom resources, the `Cluster` variable allows you to filter metrics specific to a particular cluster. This feature ensures that you can easily monitor or debug individual RayCluster instances without being overwhelmed by the data from all clusters.
```
